data/discovery/field/retention-period.yaml

```diff
@@ -1,6 +1,6 @@
 cardinality: '1'
-datatype: string
+datatype: datetime
```

Comment:
This should be either a `period` or `string`.  @michaelabenyohai is on top of making `period` possible.